### ISSUE TITLE
My result accuracy from validation too low for programs in a book. what's wrong?

### ISSUE THREAD

**Issue by chankim** [keras-team/keras#12396]
Please make sure that the boxes below are checked before you submit your issue.
If your issue is an **implementation question**, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [on the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) instead of opening a GitHub issue.

Thank you!

- [ V] Check that you are up-to-date with the master branch of Keras. You can update with:
`pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps`

- [V ] Check that your version of TensorFlow is up-to-date. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ V] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

Hi, I was learning keras/tensorflow follwing the book 'Deep Learning with Python - Develop Deep Learning Models on Theano and TensorFlow Using Keras' by Jason Brownlee. In some recent chapters, I'm seeing low accuracy that expected in the book. For example, in chapter 10, I ran this code (in python 3.5 virtual environment).

```
# Multiclass Classification with the Iris Flowers Dataset
import numpy
from pandas import read_csv
from keras.models import Sequential
from keras.layers import Dense
from keras.wrappers.scikit_learn import KerasClassifier
from keras.utils import np_utils
from sklearn.model_selection import cross_val_score
from sklearn.model_selection import KFold
from sklearn.preprocessing import LabelEncoder
from sklearn.pipeline import Pipeline
# fix random seed for reproducibility
seed = 7
numpy.random.seed(seed)
# load dataset
dataframe = read_csv("iris.csv", header=None)
dataset = dataframe.values
X = dataset[:,0:4].astype(float)
Y = dataset[:,4]
# encode class values as integers
encoder = LabelEncoder()
encoder.fit(Y)
encoded_Y = encoder.transform(Y)
# convert integers to dummy variables (i.e. one hot encoded)
dummy_y = np_utils.to_categorical(encoded_Y)
# define baseline model
def baseline_model():
    # create model
    model = Sequential()
    model.add(Dense(4, input_dim=4, init='normal', activation='relu'))
    model.add(Dense(3, init='normal', activation='sigmoid'))
    # Compile model
    model.compile(loss='categorical_crossentropy', optimizer='adam', metrics=['accuracy'])
    return model
estimator = KerasClassifier(build_fn=baseline_model, nb_epoch=200, batch_size=5, verbose=0)
kfold = KFold(n_splits=10, shuffle=True, random_state=seed)
results = cross_val_score(estimator, X, dummy_y, cv=kfold)
print("Accuracy: %.2f%% (%.2f%%)" % (results.mean()*100, results.std()*100))
```

And the result is this (The book says it should be about 95% but I see only 35%):
```
(venv3) ckim@chan-ubuntu:~/Tensorflow/chapter_10$ python iris_example.py
Using TensorFlow backend.
iris_example.py:30: UserWarning: Update your `Dense` call to the Keras 2 API: `Dense(4, kernel_initializer="normal", activation="relu", input_dim=4)`
  model.add(Dense(4, input_dim=4, init='normal', activation='relu'))
WARNING:tensorflow:From /home/ckim/venv3/lib/python3.5/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
iris_example.py:31: UserWarning: Update your `Dense` call to the Keras 2 API: `Dense(3, kernel_initializer="normal", activation="sigmoid")`
  model.add(Dense(3, init='normal', activation='sigmoid'))
WARNING:tensorflow:From /home/ckim/venv3/lib/python3.5/site-packages/tensorflow/python/ops/math_ops.py:3066: to_int32 (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
2019-03-05 17:18:50.431431: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 AVX512F FMA
2019-03-05 17:18:50.820254: I tensorflow/compiler/xla/service/service.cc:150] XLA service 0x5b88780 executing computations on platform CUDA. Devices:
2019-03-05 17:18:50.820310: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (0): Quadro GV100, Compute Capability 7.0
2019-03-05 17:18:50.820326: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (1): GeForce GTX 1050, Compute Capability 6.1
2019-03-05 17:18:50.843118: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 3100000000 Hz
2019-03-05 17:18:50.845473: I tensorflow/compiler/xla/service/service.cc:150] XLA service 0x5c715e0 executing computations on platform Host. Devices:
2019-03-05 17:18:50.845516: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (0): <undefined>, <undefined>
2019-03-05 17:18:50.846360: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1433] Found device 0 with properties: 
name: Quadro GV100 major: 7 minor: 0 memoryClockRate(GHz): 1.627
pciBusID: 0000:b3:00.0
totalMemory: 31.72GiB freeMemory: 31.31GiB
2019-03-05 17:18:50.846479: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1433] Found device 1 with properties: 
name: GeForce GTX 1050 major: 6 minor: 1 memoryClockRate(GHz): 1.455
pciBusID: 0000:65:00.0
totalMemory: 1.95GiB freeMemory: 1.24GiB
2019-03-05 17:18:50.846523: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1497] Ignoring visible gpu device (device: 1, name: GeForce GTX 1050, pci bus id: 0000:65:00.0, compute capability: 6.1) with Cuda multiprocessor count: 5. The minimum required count is 8. You can adjust this requirement with the env var TF_MIN_GPU_MULTIPROCESSOR_COUNT.
2019-03-05 17:18:50.846543: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1512] Adding visible gpu devices: 0
2019-03-05 17:18:50.851353: I tensorflow/core/common_runtime/gpu/gpu_device.cc:984] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-03-05 17:18:50.851383: I tensorflow/core/common_runtime/gpu/gpu_device.cc:990]      0 1 
2019-03-05 17:18:50.851398: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1003] 0:   N N 
2019-03-05 17:18:50.851411: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1003] 1:   N N 
2019-03-05 17:18:50.852190: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 30456 MB memory) -> physical GPU (device: 0, name: Quadro GV100, pci bus id: 0000:b3:00.0, compute capability: 7.0)
2019-03-05 17:18:51.677055: I tensorflow/stream_executor/dso_loader.cc:152] successfully opened CUDA library libcublas.so.10.0 locally
Accuracy: 32.67% (13.15%)
```

I think you can get the iris.csv file from https://gist.github.com/netj/8836201

My installation  info : 
(venv3) ckim@chan-ubuntu:~/Tensorflow/chapter_10$ pip show tensorflow-gpu
Name: tensorflow-gpu
Version: 1.13.1
Summary: TensorFlow is an open source machine learning framework for everyone.
Home-page: https://www.tensorflow.org/
Author: Google Inc.
Author-email: opensource@google.com
License: Apache 2.0
Location: /home/ckim/venv3/lib/python3.5/site-packages
Requires: numpy, astor, keras-applications, termcolor, wheel, absl-py, tensorboard, gast, keras-preprocessing, grpcio, tensorflow-estimator, protobuf, six
Required-by: 

(venv3) ckim@chan-ubuntu:~/Tensorflow/chapter_10$ pip show keras
Name: Keras
Version: 2.2.4
Summary: Deep Learning for humans
Home-page: https://github.com/keras-team/keras
Author: Francois Chollet
Author-email: francois.chollet@gmail.com
License: MIT
Location: /home/ckim/venv3/lib/python3.5/site-packages
Requires: six, keras-preprocessing, keras-applications, pyyaml, h5py, numpy, scipy
Required-by:
